app/app.go

```diff
@@ -276,10 +248,17 @@ func (d *Deployments) handleArtifact(ctx context.Context,
 					artifactID, err)
 			}
 		}()
-		return "", errors.Wrap(
-			ErrInvalidArtifactSize,
-			"size smaller than expected",
-		)
+		return "", ErrModelArtifactFileTooLarge
+	}
+	// close the pipe
+	pW.Close()
+
+	// Assign artifact size to the actual uploaded size
+	multipartUploadMsg.ArtifactSize = MaxImageSize - lr.N
```

Comment:
are we *sure* here that `MaxImageSize>=lr.N` ?

Reply:
Yes, the `else if` above catches that case.